### ISSUE TITLE
Event fired twice in shadow DOM and only once in shady DOM.

### ISSUE THREAD

**Issue by blasten** [Polymer/polymer#3719]
A user reported this issue in iron-list. For context, iron-list has a property named `selectedItem` that notifies. Inside iron-list there's an `array-selector` element which also happens to have a property `selectedItem` that notifies. Those two properties are two-way bound. https://github.com/PolymerElements/iron-list/blob/master/iron-list.html#L186.

Using native shadow DOM, the event `selected-item-changed` fires twice whereas in shady it only fires once. @azakus mentioned to me that there's code in Polymer to reuse the same event for shady which might be causing this issue. I'd like that event to only fire once, but I'm not sure if given the current setup the correct behavior is to fire the event twice.

> From https://github.com/PolymerElements/iron-list/issues/222

`selected-item-changed` event is fired multiple times when an item is selected or deselected.

`selected-items-changed` seems to be working correctly.

JSBin: http://jsbin.com/hivepag/edit?html,console,output

Tried all major version of `iron-list`, seems to affect all. Tested on OS X Chrome 49. 

R: @sorvell

**Comment by kaste** [comment]
Duplicate of #3403 and #3226

**Comment by TimvdLippe** [comment]
This seems to be fixed now: http://jsbin.com/fimidizefe/edit?html,console,output Clicking in order foo, bar, baz the output for both shadow and shady dom is
```js
[object Object] {
  queueProperty: true,
  value: "foo"
}
[object Object] {
  queueProperty: true,
  value: null
}
[object Object] {
  queueProperty: true,
  value: "bar"
}
[object Object] {
  queueProperty: true,
  value: null
}
[object Object] {
  queueProperty: true,
  value: "baz"
}
```

**Comment by dman777** [comment]
I am having this issue in Polymer 3 with

```
              <paper-dropdown-menu
                selected="{{foo}}"
                no-animations
                on-selected-item-changed="_determineSelection"
                label="min"
                name="foo">
```
using shadow dom/chrome.
